src/decouplededitor.js

```diff
@@ -57,23 +55,51 @@ export default class DecoupledEditor extends Editor {
 	 * {@link module:editor-decoupled/decouplededitor~DecoupledEditor.create `DecoupledEditor.create()`} method instead.
 	 *
 	 * @protected
-	 * @param {HTMLElement|String} sourceElementOrData The DOM element that will be the source for the created editor
+	 * @param {Object.<String,String>|Object.<String,HTMLElement>} sourceElementsOrData The object where each key
+	 * is the name of the root and corresponding value is the DOM element that will be the source for the created editor
 	 * (on which the editor will be initialized) or initial data for the editor. For more information see
-	 * {@link module:editor-balloon/ballooneditor~BalloonEditor.create `BalloonEditor.create()`}.
+	 * {@link module:editor-decoupled/decouplededitor~DecoupledEditor.create `DecoupledEditor.create()`}.
 	 * @param {module:core/editor/editorconfig~EditorConfig} config The editor configuration.
 	 */
-	constructor( sourceElementOrData, config ) {
+	constructor( sourceElementsOrData, config ) {
 		super( config );
 
-		if ( isElement( sourceElementOrData ) ) {
-			this.sourceElement = sourceElementOrData;
-		}
+		/**
+		 * The array of all sourceElements (which becomes editable areas) if any were passed in `sourceElementsOrData`.
+		 *
+		 * @protected
+		 * @member {Array.<HTMLElement>}
+		 */
+		this._sourceElements = [];
+
+		/**
+		 * The array of all root names.
+		 *
+		 * @protected
+		 * @member {Array.<String>}
+		 */
+		this._rootNames = sourceElementsOrData ? Object.keys( sourceElementsOrData ) : [];
```

Comment:
I do not like `: [];` part. It does not make much sense to create the editor with no editable roots. You should throw if threre is no roots. This parameters wasn't optional, was it?

Reply:
That's true, it wasn't optional, but there weren't any checks for it or errors defined. So probably the code should assume it is always present as defined in the constructor docs?

Reply:
Yes. In general, we do not check if parameters provided to the method are correct. We do it in some cases, where we expect that user can make a mistake, but I think this is not that case.